pkg/git/k8s_git_keychain.go

```diff
@@ -28,15 +29,40 @@ func newK8sGitKeychain(k8sClient k8sclient.Interface) *k8sGitKeychain {
 }
 
 func (k *k8sGitKeychain) Resolve(namespace, serviceAccount string, git v1alpha1.Git) (transport.AuthMethod, error) {
-	creds, err := k.secretManager.SecretForServiceAccountAndURL(serviceAccount, namespace, git.URL)
-	if err != nil && !k8serrors.IsNotFound(err) {
+	endpoint, err := transport.NewEndpoint(git.URL)
+	if err != nil {
 		return nil, err
 	}
-	if k8serrors.IsNotFound(err) {
-		return anonymousAuth, nil
-	}
 
-	return &http.BasicAuth{Username: creds.Username, Password: creds.Password}, nil
+	switch endpoint.Protocol {
```

Comment:
Similar to the Mounted Secret keychain can we just have the secretManager return all the secrets on a service account and iterate over them in here?

Reply:
I agree. This is a remnant of us revisiting the actual design of the solution. I kinda put up a PR for review as soon as I got it working cuz I didn't have the energy left to clean things up.